src/base/utils/looks_like.jl

```diff
@@ -0,0 +1,88 @@
+"""
+    looks_like_exchange_reaction(rxn_id::String;
+        exclude_biomass = false,
+        biomass_strings = _constants.biomass_strings,
+        ex_prefixes = _constants.exchange_prefixes,
+    )
+
+A predicate function that can be used to identify reactions that are probably 
+exchange or biomass reactions. Exchange reactions are identified based on matching prefixes 
+in the set `ex_prefixes` and biomass reactions are identified by looking for occurences 
+of `biomass_strings` in the reaction id.
+
+Note: the order of the probable exchange reactions identified here does not necessarily match 
+that of the exchange metabolites identified in [`looks_like_exchange_metabolite`](@ref).
+
+# Example
+```
+filter(looks_like_exchange_reaction, reactions(model)) # returns strings
+findall(looks_like_exchange_reaction, reactions(model)) # returns indices
+
+# to use the optional arguments you need to instantiate the function
+filter(x -> looks_like_exchange_reaction(x; exclude_biomass=true), reactions(model)) # returns strings
+findall(x -> looks_like_exchange_reaction(x; exclude_biomass=true), reactions(model)) # returns indices
+```
+"""
+function looks_like_exchange_reaction(rxn_id::String;
```

Comment:
simplify logic, now the whole condition looks like, schematically, after removing the redundant returns:
```
if eb
	anyprefix && !anybiomass
else
	anyprefix
```

you may convert the if to logic:
```
(eb && anyprefix && !anybiomass)
 ||
(!eb && anyprefix)
```
and factor out the anyprefix:
```
anyprefix && (eb && !anyboimass  || !eb)
```

further, the `eb &&` in the parens is now redundant because either the whole paren is true (`!eb`) or doesn't depend on value of `eb` (`!!eb`). So finally,

```
anyprefix && (!eb || !anybiomass)
```

but whoa, this may shortcut better:

```
anyprefix && !(eb && anybiomass)
```

Reply:
I want to heart this again ❤️